### PR TITLE
feat(integrations): add a `sep` argument to modify the metric name format (lightgbm, xgboost)

### PR DESCRIPTION
**Pull request by i-aki-y**
Description
-----------
What does the PR do?

In the xgboost and lightgbm integration, the metric name is constructed by the hard coded format f"{data}-{metrics}" or "{data}_{metrics}". But it is useful if we can specify the separator like "/" because the wandb recognizes "/" as a section separator and we can make plots neat and tidy.

Testing
-------
How was this PR tested?

I actually ran the following simple example and got the expected results on the wandb page.
The image is an actual result where the new sections ('xgb_valid', 'lgb_valid') were automatically created.

I am not sure how to write unit tests for this type of feature, but I would appreciate any suggestion or advice about that.

```py
from sklearn.datasets import load_iris
from sklearn.model_selection import train_test_split
import xgboost as xgb
import lightgbm as lgb
import wandb.integration.xgboost as xgb_integ
import wandb.integration.lightgbm as lgb_integ
import wandb

wandb.init(project="dev-wandb")

data = load_iris()
X_train, X_valid, y_train, y_valid = train_test_split(data['data'], data['target'], test_size=.2)

# xgb
xgb_train = xgb.DMatrix(X_train, y_train)
xgb_valid = xgb.DMatrix(X_valid, y_valid)
xgb_params = dict(max_depth=2, learning_rate=1, objective='multi:softmax', num_class=3, eval_metric="mlogloss")
xgb.train(
    xgb_params,
    xgb_train,
    evals=[(xgb_valid, "xgb_valid"), (xgb_train, "xgb_train")],
    num_boost_round=10,
    callbacks=[xgb_integ.WandbCallback()]
)

# lgb
lgb_params = dict(max_depth=2, learning_rate=1, objective='multiclass', num_class=3, metric="multi_logloss")
lgb_train = lgb.Dataset(X_train, y_train)
lgb_valid = lgb.Dataset(X_valid, y_valid)
lgb.train(
    lgb_params,
    lgb_train,
    valid_sets=[lgb_train, lgb_valid],
    valid_names=["lgb_train", "lgb_valid"],
    num_boost_round=10,
    callbacks=[lgb_integ.wandb_callback(), lgb.log_evaluation(period=1)],
)
```

wandb_plot

Checklist
-------
- [ ] Include reference to internal ticket "Fixes WB-NNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
- [x] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/main/CONTRIBUTING.md#conventional-commits)

### CODE REVIEW

**Comment by morganmcg1** [comment]
Thanks for the PR! @ayulockin could you take a look here please?

**Review by ayulockin — CHANGES_REQUESTED** [review]

**Comment by ayulockin** [comment]
Hey @i-aki-y the changes look good to me. I have asked for a few minor changes. Can you also ensure that the tests are passing?

**Comment by i-aki-y** [comment]
@ayulockin Thank you for your review. I updated the code and docs. And I have confirmed that the default behaviors have correctly changed in my environment.

File: wandb/integration/lightgbm/__init__.py [hunk 1]
```diff
@@ -94,12 +94,15 @@ def _log_feature_importance(model: "Booster") -> None:
     )
 
 
-def wandb_callback(log_params: bool = True, define_metric: bool = True) -> Callable:
+def wandb_callback(
+    log_params: bool = True, define_metric: bool = True, sep: str = "/"
```
Comment (kptkin) [inline wandb/integration/lightgbm/__init__.py:98]: My only question is should we keep backward computability here? i.e. sep="_" and add in the documentation comment that changing it to "/" will create the hierarchical view? 

I don't have strong feelings here and open to hear what others think

File: wandb/integration/xgboost/xgboost.py [hunk 2]
```diff
@@ -60,6 +60,7 @@ class WandbCallback(xgb.callback.TrainingCallback):
         log_feature_importance: (boolean) if True log a feature importance bar plot
         importance_type: (str) one of {weight, gain, cover, total_gain, total_cover} for tree model. weight for linear model.
         define_metric: (boolean) if True (default) capture model performance at the best step, instead of the last step, of training in your `wandb.summary`.
+        sep: (str) separator that combines the validation set name and metric name. (default: "/")
```
Comment (kptkin) [inline wandb/integration/xgboost/xgboost.py:63]: same as above

**Comment by i-aki-y** [comment]
@kptkin Thank you for your review.

@ayulockin What do you think about this?

I think the "/" is a good default separator, but if backward compatibility should be taken into account, keeping original separators ("-" or "_") is less surprising to the existing users.

The important thing is to parametrize the separator so I will follow the community's policy about changing the default values.

Let me know if I should fix the default values.
